test/built-ins/TypedArray/prototype/subarray/speciesctor-get-ctor-inherited.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.subarray
+description: get inherited constructor on SpeciesConstructor
+info: >
+  22.2.3.27 %TypedArray%.prototype.subarray( begin , end )
+
+  ...
+  17. Return ? TypedArraySpeciesCreate(O, argumentsList).
+
+  22.2.4.7 TypedArraySpeciesCreate ( exemplar, argumentList )
+
+  ...
+  3. Let constructor be ? SpeciesConstructor(exemplar, defaultConstructor).
+  ...
+
+  7.3.20 SpeciesConstructor ( O, defaultConstructor )
+
+  1. Assert: Type(O) is Object.
+  2. Let C be ? Get(O, "constructor").
+  3. If C is undefined, return defaultConstructor.
+  ...
+includes: [testTypedArray.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA([40, 41, 42, 43]);
+  var calls = 0;
+  var result;
+
+  Object.defineProperty(TA.prototype, "constructor", {
+    get: function() {
+      calls++;
+    }
+  });
+
+  result = sample.subarray(0);
+
+  assert.sameValue(calls, 1, "called custom ctor get accessor once");
+
+  assert.sameValue(
+    Object.getPrototypeOf(result),
+    Object.getPrototypeOf(sample),
+    "use defaultCtor on an undefined return - getPrototypeOf check"
+  );
+  assert.sameValue(
+    result.constructor,
+    undefined,
+    "used defaultCtor but still checks the inherited .constructor"
```

Comment:
I fixed this in 1091944

> Add an assertion using hasOwnProperty 

That wouldn't be very clear as this file is testing the Get("constructor") on an inherited property.

>  Add an assertion that calls is now 2

I did it on a slightly different approach, setting a custom value to calls and than checking a get on `result.constructor` to verify it is using that property.

> Define the constructor accessor property on the sample instance--not the TA prototype

That's the scope for `speciesctor-get-ctor` only.